pkg/operator/webhookconfig.go

```diff
@@ -1,84 +1,121 @@
+// be nice if the CVO could apply the configuration as it is mostly static.
+// Unfortunately, the CA bundle is not known until runtime.
+type WebhookConfigUpdater struct {
 	caPath string
-	client *admissionregistrationv1beta1.AdmissionregistrationV1beta1Client
+	client client.Client
 }
 
-// NewWebhookCAUpdater returns a new WebhookCAUpdater.
-func NewWebhookCAUpdater(mgr manager.Manager, caPath string) (*WebhookCAUpdater, error) {
-	var err error
-
-	w := &WebhookCAUpdater{caPath: caPath}
-
-	w.client, err = admissionregistrationv1beta1.NewForConfig(mgr.GetConfig())
-	if err != nil {
-		return nil, err
+// NewWebhookConfigUpdater returns a new WebhookConfigUpdater instance.
+func NewWebhookConfigUpdater(mgr manager.Manager, caPath string) (*WebhookConfigUpdater, error) {
+	w := &WebhookConfigUpdater{
+		caPath: caPath,
+		client: mgr.GetClient(),
 	}
 
 	return w, nil
 }
 
-// Start fetches the current CA bundle from disk and sets it on the webhook
-// configurations, then simply waits for the stop channel to close.
-func (w *WebhookCAUpdater) Start(stopCh <-chan struct{}) error {
-	ca, err := w.GetEncodedCA()
-	if err != nil {
-		return err
-	}
-
-	// TODO: This should probably replace the caBundle in all webhook client
-	// configurations in the object, but unfortuntaely that's not easy to do
-	// with a JSON patch.  For now this only modifies the first entry.
-	patch := []jsonpatch.Operation{
-		{
-			Operation: "replace",
-			Path:      "/webhooks/0/clientConfig/caBundle",
-			Value:     ca,
+// Start creates or updates the webhook configurations then waits for the stop
+// channel to be closed.
+func (w *WebhookConfigUpdater) Start(stopCh <-chan struct{}) error {
+	vc := &admissionregistrationv1beta1.ValidatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "admissionregistration.k8s.io/v1beta1",
+			Kind:       "ValidatingWebhookConfiguration",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: WebhookConfigurationName,
+			Labels: map[string]string{
+				"k8s-app": OperatorName,
+			},
 		},
 	}
 
-	patchBytes, err := json.Marshal(patch)
-	if err != nil {
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), w.client, vc, func() error {
+		var err error
+		vc.Webhooks, err = w.ValidatingWebhooks()
 		return err
-	}
-
-	_, err = w.client.ValidatingWebhookConfigurations().
-		Patch(WebhookConfigurationName, types.JSONPatchType, patchBytes)
+	})
 
 	if err != nil {
 		return err
 	}
 
-	klog.Info("Updated webhook configuration CA certificates.")
+	klog.Infof("Webhook configuration status: %s", op)
 
 	// Block until the stop channel is closed.
 	<-stopCh
 
 	return nil
 }
 
+// ValidatingWebhooks returns the validating webhook configurations.
+func (w *WebhookConfigUpdater) ValidatingWebhooks() ([]admissionregistrationv1beta1.Webhook, error) {
+	caBundle, err := w.GetEncodedCA()
+	if err != nil {
+		return nil, err
+	}
+
+	failurePolicy := admissionregistrationv1beta1.Ignore
```

Comment:
Why should this be ignore?

Reply:
Because the validations are really (or will be when we're not doing allow-all) a nice-to-have. We have OpenAPI validation on the fields where we're able to do that, and other errors will be reported in events or in status conditions after the resource is created. So, we shouldn't completely block the functioning of the operator if the connection to the webhook server is not working -- e.g. if there's some transient networking issue.